### PR TITLE
[Mobile app] Enhancing the insights graphs

### PR DESCRIPTION
**Pull request by NoahNsimbe**
#### Summary of Changes (What does this PR do?)

- [x] Configures state management on the insights page.
- [x] Fixes forecast toggle bug.
- [x] Migrates the database from native [Sqflite](https://pub.dev/packages/sqflite) to [Drift](https://docs.flutter.dev/development/packages-and-plugins/happy-paths/recommended#structured-local-storage-).

#### Is this change ready to hit production in its current state?
- [x] Yes
- [ ] No

#### Status of maturity (all need to be checked before merging):

- [x] I consider this code done
- [x] I've tested this locally
- [ ] This branch has been LGTM'ed by a reviewer

#### How should this be manually tested?
* Please include the steps to be done inorder to setup and test this PR.

#### What are the relevant tickets?
- [MOB-118](https://airqoteam.atlassian.net/browse/MOB-118)

#### Screenshots (optional)

### CODE REVIEW

File: mobile/lib/blocs/insights/insights_bloc.dart [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+import 'package:charts_flutter/flutter.dart' as charts;
+import 'package:equatable/equatable.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+import '../../constants/config.dart';
+
+part 'insights_event.dart';
+part 'insights_state.dart';
+
+class InsightsBloc extends Bloc<InsightsEvent, InsightsState> {
```
Comment (NoahNsimbe) [inline mobile/lib/blocs/insights/insights_bloc.dart:15]: State management for insights page

File: mobile/lib/models/database.dart [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  TextColumn get siteId => text()();
+  IntColumn get frequency => intEnum<Frequency>()();
+
+  @override
+  Set<Column> get primaryKey => {
+        siteId,
+        frequency,
+        time,
+      };
+}
```
Comment (NoahNsimbe) [inline mobile/lib/models/database.dart:26]: Insight (measurements) model

File: mobile/lib/models/database.dart [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+class HistoricalInsights extends Insight {}
+
+@DataClassName("ForecastInsight")
+class ForecastInsights extends Insight {}
+
+@DriftDatabase(tables: [HistoricalInsights, ForecastInsights])
+class AirQoDatabase extends _$AirQoDatabase {
+  factory AirQoDatabase() {
+    return _instance;
+  }
```
Comment (NoahNsimbe) [inline mobile/lib/models/database.dart:38]: Registering historical and forecast data tables

File: mobile/lib/models/database.dart (hunk 2 above)
Comment (Baalmart) [reply mobile/lib/models/database.dart:26]: Thanks @NoahNsimbe for this. I will just leave a few file organisational questions for your review.

I sense that the insights.dart file has been deleted in favor of the database.dart file.

1. What is the main motivation for this rename/restructure?
2. What is the organisation structure here? Is the `database` file supposed to represent an entity?

File: mobile/lib/models/chart_data.dart [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+class ChartData {
+  const ChartData({
+    required this.frequency,
+    required this.available,
+    required this.dateTime,
+    required this.siteId,
+    required this.pm2_5,
+    required this.pm10,
+  });
+
```
[the remark this answers is not in the document]
Comment (Baalmart) [reply mobile/lib/models/chart_data.dart:14]: Thanks!

**Review by Baalmart — COMMENTED** [review]
Thanks @NoahNsimbe for this. Just left a few comments.

Also, since we are on the topic of insights.....there is this old feedback item around inconsistencies with values on the insights page and those on the home timeline. I hop we consider making that content consistent OR explicitly informing the user that the latest air quality on the 2 sections is different by design.

**Comment by NoahNsimbe** [comment]
> Thanks @NoahNsimbe for this. Just left a few comments.
> 
> Also, since we are on the topic of insights.....there is this old feedback item around inconsistencies with values on the insights page and those on the home timeline. I hop we consider making that content consistent OR explicitly informing the user that the latest air quality on the 2 sections is different by design.

This has been resolved in this PR.

**Comment by NoahNsimbe** [comment]
**What is the main motivation for this rename/restructure?**
The insights entity does not need to have its own file. It's the only entity using the relational database, we can have it as part of the database file. 

**What is the organisation structure here? Is the database file supposed to represent an entity?**
For now, the database file handles relational entities.  We have only one relational entity. We shall refactor when we get more entities that use the relational database.

@Baalmart

**Review by Baalmart — APPROVED** [review]
Thanks @NoahNsimbe , this LGTM

**Comment by Baalmart** [comment]
Hi @NoahNsimbe , the not-ready-for-merge flag is somehow confusing.....
Could we perhaps just use two flags, that is:

1. work-in-progress
2. ready-for-review

This recent usage of ready for merge and ready for review is quite new and can easily get out of hand since it could encourage the introduction of newer confusing labels which could have the same motivation.

**Comment by NoahNsimbe** [comment]
> Hi @NoahNsimbe , the not-ready-for-merge flag is somehow confusing..... Could we perhaps just use two flags, that is:
> 
> 1. work-in-progress
> 2. ready-for-review
> 
> This recent usage of ready for merge and ready for review is quite new and can easily get out of hand since it could encourage the introduction of newer confusing labels which could have the same motivation.

Updated the flags @Baalmart

**Review by Mnoble-19 — APPROVED** [review]
no-forecast
forecast

**Review by Baalmart — COMMENTED** [review]
Hi @NoahNsimbe , this PR has a couple of merge conflicts which need to be resolved. 
Thanks a lot, this is good work. cc: @BenjaminSsempala

**Pull request merged**